deploy/eck-operator/templates/_helpers.tpl

```diff
@@ -67,17 +67,53 @@ Create the name of the service account to use
 {{- end }}
 {{- end }}
 
+{{/*
+Determine effective Kubernetes version
+*/}}
+{{- define "eck-operator.effectiveKubeVersion" -}}
+{{- if .Values.internal.manifestGen -}}
+{{- semver .Values.internal.kubeVersion -}}
+{{- else -}}
+{{- .Capabilities.KubeVersion.Version -}}
+{{- end -}}
+{{- end -}}
+
 {{/*
 Add the webhook sideEffects field on supported Kubernetes versions
 */}}
 {{- define "eck-operator.webhookSideEffects" -}}
-{{- $kubeVersion := .Capabilities.KubeVersion.Version -}}
+{{- $kubeVersion := (include "eck-operator.effectiveKubeVersion" .) -}}
 {{- $kubeVersionSupported := semverCompare ">=1.13.0-0" $kubeVersion -}}
-{{- if and $kubeVersionSupported (not .Values.internal.manifestGen) }}
+{{- if $kubeVersionSupported }}
 sideEffects: "None"
 {{- end }}
 {{- end }}
 
+{{/*
+Use v1 of ValidatingWebhookConfiguration on supported Kubernetes versions
+*/}}
+{{- define "eck-operator.webhookAPIVersion" -}}
+{{- $kubeVersion := (include "eck-operator.effectiveKubeVersion" .) -}}
+{{- $kubeVersionSupported := semverCompare ">=1.16.0-0" $kubeVersion -}}
+{{- if $kubeVersionSupported -}}
+admissionregistration.k8s.io/v1
+{{- else -}}
+admissionregistration.k8s.io/v1beta1
+{{- end -}}
+{{- end }}
+
+
+{{/*
+Define admissionReviewVersions based on Kubernetes version
+*/}}
+{{- define "eck-operator.webhookAdmissionReviewVersions" -}}
+{{- $kubeVersion := (include "eck-operator.effectiveKubeVersion" .) -}}
+{{- $kubeVersionSupported := semverCompare ">=1.16.0-0" $kubeVersion -}}
+{{- if $kubeVersionSupported  }}
+admissionReviewVersions: [v1beta1, v1]
```

Comment:
Maybe I am misunderstanding how this works but the operator does not currently offer a v1 webhook. Should we not just list `v1beta1` here?

Reply:
I kept it that way because that's what Kubebuilder generates (even though it doesn't v1 either). But, you're right. It's probably confusing. I'll remove it.